addons/mail/static/src/models/composer_view/composer_view.js

```diff
@@ -300,105 +308,43 @@ registerModel({
-                if (composer.thread.model === 'mail.channel') {
-                    Object.assign(postData, {
-                        subtype_xmlid: 'mail.mt_comment',
-                    });
-                } else {
-                    Object.assign(postData, {
-                        subtype_xmlid: composer.isLog ? 'mail.mt_note' : 'mail.mt_comment',
-                    });
-                    if (!composer.isLog) {
-                        params.context = { mail_post_autofollow: true };
-                    }
-                }
-                if (this.threadView && this.threadView.replyingToMessageView && this.threadView.thread !== this.messaging.inbox) {
-                    postData.parent_id = this.threadView.replyingToMessageView.message.id;
-                }
-                const { threadView = {} } = this;
-                const { thread: chatterThread } = this.chatter || {};
-                const { thread: threadViewThread } = threadView;
-                const messageData = await this.env.services.rpc({ route: `/mail/message/post`, params });
-                if (!this.messaging) {
-                    return;
-                }
-                const message = this.messaging.models['Message'].insert(
-                    this.messaging.models['Message'].convertData(messageData)
-                );
-                for (const threadView of message.originThread.threadViews) {
-                    // Reset auto scroll to be able to see the newly posted message.
-                    threadView.update({ hasAutoScrollOnMessageReceived: true });
-                }
-                if (chatterThread) {
-                    if (this.exists()) {
-                        this.delete();
-                    }
-                    if (chatterThread.exists()) {
-                        chatterThread.fetchData(['followers', 'suggestedRecipients']);
-                    }
-                }
-                if (threadViewThread) {
-                    if (threadViewThread === this.messaging.inbox) {
-                        if (this.exists()) {
-                            this.delete();
-                        }
-                        this.env.services['notification'].notify({
-                            message: _.str.sprintf(this.env._t(`Message posted on "%s"`), message.originThread.displayName),
-                            type: 'info',
-                        });
-                    }
-                    if (threadView && threadView.exists()) {
-                        threadView.update({ replyingToMessageView: clear() });
-                    }
-                }
-                if (composer.exists()) {
-                    composer._reset();
-                }
-            } finally {
-                if (composer.exists()) {
-                    composer.update({ isPostingMessage: false });
-                }
+            this.composer.activeThread.sender.queueMessage(composition);
+            this.composer._reset();
```

Comment:
This is wrong. `MessageComposition` is a message being composed. When there is a composer view (and maybe when the content is non-empty), it is expected to have a `MessageComposition` related to this part of the UI.

Here the record is created on clicking on post, which is more a "queuing-posting" `MessageComposition`. Don't rename the model but change this code so that it flags the `MessageComposition` as to be "queuing-posting". The `MessageComposition` should be created much sooner than user interaction to post the message composition.

Reply:
That is not how I see it.
The purpose of this PR is to remove the sending part from the composer by storing away the message data in a message queue and send them at there own pace. I don't think we should increase the complexity by creating message composition when it's not useful at all. 

In other words: message compositions are message ready to be send to the server. Not some way to store the data from the composer, they should not server 2 different purposes IMHO.

Reply:
Side not, I m not a fan of the name "MessageComposition" because it doesn't show the fact that those models are temporary/

Reply:
`MessageComposition` is the better term I can find for what we define. Even without the consideration that, for me, the message composition should exist much sooner than when the user click on "Send".

But honestly, this is not super important for now. But keep in mind `MessageComposition` is a message being composed, and `Composer` is what let us composer a message, so this shows that there are some cleaning that is planned for later. The right direction is introducing `MessageComposition` and clean the concept `Composer` that is poorly defined in the current code, but that's quite out-of-scope of this PR (the cleaning part of `Composer`).

Reply:
@alexkuhn What do you think about this naming:

- Temporary message in the sending queue: MessageComposed.
- Message inside the composer: MessageComposition.

I don't like to have too much model for temporary messages, but it look like what you want.